basic_autonomy/include/basic_autonomy/basic_autonomy.h

```diff
@@ -263,6 +263,14 @@ namespace basic_autonomy
      std::vector<PointSpeedPair> create_lanechange_geometry(const cav_msgs::Maneuver &maneuver, double max_starting_downtrack,
                                                                    const carma_wm::WorldModelConstPtr &wm, cav_msgs::VehicleState &ending_state_before_buffer,
                                                                    const cav_msgs::VehicleState &state, const DetailedTrajConfig &detailed_config);
+     
+     std::vector<lanelet::BasicPoint2d> create_lanechange_route(int ending_lane_id, double starting_downtrack, double ending_downtrack,
+                                                            const carma_wm::WorldModelConstPtr &wm,const cav_msgs::VehicleState &state);
+
+     cav_msgs::VehicleState get_nearest_state_to_downtrack(const std::vector<lanelet::BasicPoint2d>& points, double target_downtrack, 
+                                                            const carma_wm::WorldModelConstPtr &wm);   
+
+     std::vector<std::vector<lanelet::BasicPoint2d>> resample_linestrings_to_same_length(std::vector<lanelet::BasicPoint2d> line_1, std::vector<lanelet::BasicPoint2d> line_2);
```

Comment:
This is also kind of unclear in terms of what it refers to, is length spatial (meters) or list size (number of elements). I'd recommend clarifying by renaming to "...same_num_points" or something if that's the case.